block/local/adapter.go

```diff
@@ -127,7 +140,28 @@ func (l *Adapter) Remove(obj block.ObjectPointer) error {
 		return err
 	}
 	p = filepath.Clean(p)
-	return os.Remove(p)
+	err = os.Remove(p)
+	if err != nil {
+		return err
+	}
+	if l.removeEmptyDir {
+		dir := filepath.Dir(p)
+		removeEmptyDirUntil(dir, l.path)
+	}
+	return nil
+}
+
+func removeEmptyDirUntil(dir string, stopAt string) {
+	for stopAt != "" && strings.HasPrefix(dir, stopAt) && dir != stopAt {
```

Comment:
Yikes... if `stopAt` doesn't end in `/` this can delete a file _next to it_.  E.g. say the block adapter is rooted at `/home/ariels/data`.  Then this will happily delete a file `/home/ariels/datadog`.
So (_also_...!) need to append a `/` to `stopAt` if it does not already end in `/`'

Reply:
added / to stopAt if needed.